File: shoring/flywheel/src/test/java/com/panopset/tests/transformer/FlywheelTemplateToTextTransformerTest.java
package com.panopset.tests.transformer;

import java.io.File;
import java.io.StringWriter;

import com.panopset.flywheel.Flywheel;
import com.panopset.flywheel.FlywheelBuilder;

public class FlywheelTemplateToTextTransformerTest extends StandardTransformerTest {
	
	private final String expected;
	private final String templateFileRezPath;

	public FlywheelTemplateToTextTransformerTest(String packageName, String templateFileRezPath, final String expected) {
		super(packageName);
		this.expected = expected;
		this.templateFileRezPath = templateFileRezPath;
	}

	@Override
	protected ResultsDataSupplier createResultsDataSupplier() {
		
		return new ResultsDataSupplier() {
			
			@Override
			public String getExpectedResults() {
				return expected;
			}
			
			@Override
			public String getActualResults() {
				StringWriter sw = new StringWriter();
				File template = new StandardPackagePath(getPackageName()).getFile(templateFileRezPath);
				Flywheel script = new FlywheelBuilder().withWriter(sw).file(template)
						.construct();
				script.exec();
				return sw.toString();
			}
		};
	}
}

File: code/shoring/flywheel/src/test/kotlin/com/panopset/tests/transformer/FlywheelTemplateToTextTransformerTest.kt
package com.panopset.tests.transformer

import com.panopset.flywheel.FlywheelBuilder
import java.io.StringWriter

class FlywheelTemplateToTextTransformerTest(
    packageName: String,
    private val templateFileRezPath: String,
    private val expected: String
) : StandardTransformerTest(packageName) {
    override fun createResultsDataSupplier(): ResultsDataSupplier {
        return object : ResultsDataSupplier {
            override val expectedResults: String
                get() = expected
            override val actualResults: String
                get() {
                    val sw = StringWriter()
                    val template = StandardPackagePath(packageName).getFile(templateFileRezPath)
                    val script = FlywheelBuilder().withWriter(sw).file(template)
                        .construct()
                    script.exec()
                    return sw.toString()
                }
        }
    }
}
